fabfile.py

```diff
@@ -0,0 +1,149 @@
+        run("cp -r %s %s" % (config.atf_build_dir + "StartSDL.sh", test_run_dir))
+        run("cp -r %s %s" % (config.atf_build_dir + "StopSDL.sh", test_run_dir))
+        run("cp -r %s %s" % (config.atf_build_dir + "WaitClosingSocket.sh", test_run_dir))
+        run("cp -r %s %s" % (config.atf_build_dir + "start.sh", test_run_dir))
+        run("cp -r %s %s" % (config.atf_build_dir + "interp", test_run_dir))
+        run("cp -r %s %s" % (config.scripts_clone_dir + "/*", test_run_dir))
+        run("cp -r %s %s" % (config.sdl_build_dir + "bin", test_run_dir + "SDL_bin"))
+        logger_library_path = config.sdl_build_dir + "log4cxx_build/" + arch() + "/lib/"
+        run("cp -r %s %s" % (logger_library_path + "/*", test_run_dir + "SDL_bin/"))
+
+
+@task
+def tests_run(run_dir=config.test_run_dir):
+    if not exists(run_dir):
+        print("%s does not exists" % run_dir)
+        return 1
+    with cd(run_dir):
+        output = run('ls test_scripts')
+        scripts = output.split()
+        known_issues = run("cat KnownIssues.md")
+        known_issues = get_known_issues(known_issues)
+        new_failed = {}
+        print(scripts)
+        for script in scripts:
+            print("Execute %s" % script)
+            with settings(warn_only=True):
+                output = run('''./start.sh --storeFullSDLLogs \
+                             --sdl_core=./SDL_bin/ test_scripts/%s | tee console_output''' % script)
+                script_reports_dir = "%s/%s" % (config.reports_dir, script)
+                run("mkdir -p %s" % script_reports_dir)
+                run("mv console_output %s/" % script_reports_dir)
+                run("mv TestingReports %s/" % script_reports_dir)
+                new_failed[script] = get_list_of_failed_test_cases(output)
+                print("List of failed test cases in %s : " % script)
+                for failed_case in new_failed[script]:
+                    print("\t %s" % failed_case)
+                if script in known_issues:
+                    new_failed[script] = filter_known_issues(new_failed[script], known_issues[script])
+        print("New failed test cases:")
+        for script in new_failed:
+            failed_in_script = new_failed[script]
+            print("* %s:" % script)
+            for case in failed_in_script:
+                print("  * %s" % case)
+
+
+@task
+def setup():
+    run("sudo chmod 4755 /sbin/ifconfig")
+
+
+@task
+def prepare():
+    setup()
+    clone_sdl()
+    clone_atf()
+    clone_scripts()
+    build_sdl()
+    build_atf()
+    prepare_test_run()
+
+
+@task
+def clear():
+    run("rm -rf  %s" % config.work_dir)
+
+
+@task
+def reset():
+    run("rm -rf  %s" % config.test_run_dir)
+    prepare_test_run()
+
+
+@task
+def test():
+    with settings(warn_only=True):
+        o = run("ls /tmp/askdjfn")
+        print(o)
+        print(o.return_code)
+    o = run("ls /tmp/")
+    print(o)
+    print(o.return_code)
```

Comment:
It's good practice to use [main](https://www.artima.com/weblogs/viewpost.jsp?thread=4829).